graphql-dgs-client/src/main/kotlin/com/netflix/graphql/dgs/client/DefaultGraphQLClient.kt

```diff
@@ -54,7 +54,7 @@ class DefaultGraphQLClient(private val url: String) : GraphQLClient, MonoGraphQL
             "Content-type" to listOf("application/json")
         )
 
```

Comment:
Better to add another constructor with the additional name parameter for compatibility with java.

Reply:
Hi @srinivasankavitha, thanks for reviewing this!

Given that `Request` is private, wouldn't it only be accessible inside this class? Perhaps the behavior differs between Kotlin and Java?

Anyhow, I'll add another constructor to it. Thanks once again!

Reply:
Hi, I just wanted to double-check this. After adding the constructor, the IDE instantly tells me this constructor is never used. I also tried to use the class from Java to check what is the behavior and saw that the following would fail compilation, telling me that the constructor is private:

```java
new DefaultGraphQLClient.Companion.Request(/*...*/)
```

Thanks once again!

Reply:
Yes, you're right. Actually my comment would be more relevant for the GraphQLRequest, not so much the GraphQLClient. For the GraphQLRequest, we need 2 constructors for compatibility.

Reply:
I do question the need to update the GraphQLClient and the executeQuery methods altogether. We could just update the GraphQLRequest to take in the operation name and that should be sufficient, no? Am I missing something?

Reply:
They are different things. One is the name you give to a specific operation, and the other is the operation that you want to execute when you [send multiple operations](https://graphql.org/learn/serving-over-http/#post-request).

Take the examples below (I use `>...<` to highlight it). The first is in the query, e.g.:

```graphql
query >UserInformation< {
  me {
    id
    username
  }
}
```

The second in the actual request body ([some reasoning](https://stackoverflow.com/questions/49714484/why-would-you-send-multiple-operations-in-a-graphql-query-document-when-one-can)):

```json
{
  "operationName": ">A<",
  "query": "query A { ... }\n\nquery B { ... }",
  "variables": {}
}
```

Reply:
I see. So this means the query executor would execute only query A in the above example? Why send multiple operations at all in that case?